dom.bs

```diff
@@ -209,19 +209,59 @@ against a <var>set</var>, run these steps:
 added.
 
 
-<h3 id=namespaces>Namespaces</h3>
+<h3 id=name-validation oldids=namespaces>Name validation</h3>
 
-<p>To <dfn export>validate</dfn> a <var>qualifiedName</var>, <a>throw</a> an
-"{{InvalidCharacterError!!exception}}" {{DOMException}} if <var>qualifiedName</var> does not match
-the <code><a type>QName</a></code> production.
+<p>A [=string=] is a <dfn>valid namespace prefix</dfn> if its [=string/length=] is at least 1 and it
+does not contain [=ASCII whitespace=], U+0000 NULL, U+002F (/), or U+003E (>).
 
-<p>To <dfn export>validate and extract</dfn> a <var>namespace</var> and <var>qualifiedName</var>,
-run these steps:
+<p>A [=string=] is a <dfn>valid attribute local name</dfn> if its [=string/length=] is at least 1
+and it does not contain [=ASCII whitespace=], U+0000 NULL, U+002F (/), U+003D (=), or U+003E (>).
+
+<p>A [=string=] |name| is a <dfn>valid element local name</dfn> if the following steps return true:
 
 <ol>
- <li><p>If <var>namespace</var> is the empty string, then set it to null.
+ <li><p>If |name|'s [=string/length=] is 0, then return false.
 
- <li><p><a>Validate</a> <var>qualifiedName</var>.
+ <li>
+  <p>If |name|'s 0th [=code point=] is an [=ASCII alpha=], then:
+
+  <ol>
+   <li><p>If |name| contains [=ASCII whitespace=], U+0000 NULL, U+002F (/), or U+003E (>), then
+   return false.
+
+   <li><p>Return true.
+  </ol>
+
+ <li><p>If |name|'s 0th [=code point=] is not U+003A (:), U+005F (_), or in the range U+0080
+ to U+10FFFF, inclusive, then return false.
+
+ <li><p>If |name|'s subsequent [=code points=], if any, are not [=ASCII alphas=], [=ASCII digits=],
+ U+002D (-), U+002E (.), U+003A (:), U+005F (_), or in the range U+0080 to U+10FFFF, inclusive, then
+ return false.
+
+ <li><p>Return true.
+</ol>
+
+<p class=note>This concept is used to validate [=/element=] [=Element/local names=], when
+constructed by DOM APIs. The intention is to allow any name that is possible to construct using the
+HTML parser (the branch where the first [=code point=] is an [=ASCII alpha=]), plus some additional
+possibilities. For those additional possibilities, the ASCII range is restricted for historical
+reasons, but beyond ASCII anything is allowed.
+
+<div class=note>
+ <p>The following JavaScript-compatible regular expression is an implementation of the above
+ definition:
+
+ <pre class=lang-javascript>
+  /^(?:[A-Za-z][^\0\t\n\f\r\u0020/>]*)|(?:[:_\u0080-][A-Za-z-.:_\u0080-]*)$/u
```

Comment:
@zcorpan or @mathiasbynens could you please review this?

Reply:
Missing ASCII digits.
```suggestion
  /^(?:[A-Za-z][^\0\t\n\f\r\u0020/>]*)|(?:[:_\u0080-][A-Za-z0-9-.:_\u0080-]*)$/u
```

Reply:
I think JavaScript doesn’t support character class ranges without the end of the range being explicit, so I’d change both occurrences of `[\u0080-]` into `[\x80-\u{10FFFF}]`.

Is `[A-Za-z-.:_\u0080-]` meant to refer to step 4? In that case, it’s missing “ASCII digits”. Suggestion: `[A-Za-z0-9-.:_\x80-\u{10FFFF}]`.

- Valid namespace prefix: `[\0\t\n\f\r />]`
- Valid attribute local name: `[\0\t\n\f\r /=>]`
- Valid element local name, step 2.1: `[^\0\t\n\f\r />]`
- Valid element local name, step 3: `[:_\x80-\u{10FFFF}]`
- Valid element local name, step 4: `[A-Za-z0-9-.:_\x80-\u{10FFFF}]`

Piecing things together:

```js
/^(?:[A-Za-z][^\0\t\n\f\r\x20/>]*)|(?:[:_\x80-\u{10FFFF}][A-Za-z0-9-.:_\x80-\u{10FFFF}]*)$/u
```